Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -715,12 +715,12 @@ void FrameLoader::setOutgoingReferrer(const URL& url)
     m_outgoingReferrer = url.strippedForUseAsReferrer();
 }
 
-static String extractContentLanguageFromHeader(const String& header)
+static AtomString extractContentLanguageFromHeader(const String& header)
 {
     auto commaIndex = header.find(',');
     if (commaIndex == notFound)
-        return stripLeadingAndTrailingHTMLSpaces(header);
-    return StringView(header).left(commaIndex).stripLeadingAndTrailingMatchedCharacters(isHTMLSpace<UChar>).toString();
+        return AtomString { stripLeadingAndTrailingHTMLSpaces(header) };
```

Comment:
Slightly irritating here that if there are leading and trailing spaces we possibly make and then destroy a string that is already in the atom string table. Would be nice if there was an idiom where there are no spaces and the string is not already in the atom string table, then we put that string into the atom string table without allocating a new one, but when there are spaces to strip and the string is already in the atom string table we don’t needlessly create/destroy a temporary string.

Reply:
This isn't a regression from my change. I agree it is unfortunate that we may construct an unnecessary temporary String in some cases. I don't know yet how common it is, it may indeed be something we want to optimize if common enough.